dockers/docker-dhcp-relay/cli-plugin-tests/test_config_dhcp_relay.py

```diff
@@ -1,229 +1,265 @@
-"""
-
-config_vlan_del_dhcp_relay_output="""\
-Removed DHCP relay destination addresses ('192.0.0.100',) from Vlan1000
+config_dhcp_relay_del_output = """\
+Removed DHCP relay address [{}] from Vlan1000
 Restarting DHCP relay service...
 """
-
-config_vlan_del_dhcpv6_relay_output="""\
-Removed DHCP relay destination addresses ('fc02:2000::1',) from Vlan1000
-Restarting DHCP relay service...
-"""
-
-config_vlan_del_multiple_dhcpv6_relay_output="""\
-Removed DHCP relay destination addresses ('fc02:2000::1', 'fc02:2000::2', 'fc02:2000::3') from Vlan1000
-Restarting DHCP relay service...
-"""
-
-class TestConfigVlanDhcpRelay(object):
+expected_dhcp_relay_add_config_db_output = {
+    "ipv4": {
+        "dhcp_servers": [
+            "192.0.0.1", "192.0.0.3"]
+    },
+    "ipv6": {
+        "dhcpv6_servers": [
+            "fc02:2000::1", "fc02:2000::3"]
+    }
+}
+expected_dhcp_relay_del_config_db_output = {
+    "ipv4": {
+        "dhcp_servers": [
+            "192.0.0.1"
+        ]
+    },
+    "ipv6": {
+        "dhcpv6_servers": [
+            "fc02:2000::1"
+        ]
+    }
+}
+expected_dhcp_relay_add_multi_config_db_output = {
+    "ipv4": {
+        "dhcp_servers": [
+            "192.0.0.1", "192.0.0.3", "192.0.0.4", "192.0.0.5"
+        ]
+    },
+    "ipv6": {
+        "dhcpv6_servers": [
+            "fc02:2000::1", "fc02:2000::3", "fc02:2000::4", "fc02:2000::5"
+        ]
+    }
+}
+
+IP_VER_TEST_PARAM_MAP = {
+    "ipv4": {
+        "command": "helper",
+        "ips": [
+            "192.0.0.3",
+            "192.0.0.4",
+            "192.0.0.5"
+        ],
+        "exist_ip": "192.0.0.1",
+        "nonexist_ip": "192.0.0.2",
+        "invalid_ip": "192.0.0",
+        "table": "VLAN"
+    },
+    "ipv6": {
+        "command": "destination",
+        "ips": [
+            "fc02:2000::3",
+            "fc02:2000::4",
+            "fc02:2000::5"
+        ],
+        "exist_ip": "fc02:2000::1",
+        "nonexist_ip": "fc02:2000::2",
+        "invalid_ip": "fc02:2000:",
+        "table": "DHCP_RELAY"
+    }
+}
+
+
+@pytest.fixture(scope="module", params=["ipv4", "ipv6"])
+def ip_version(request):
+    """
+    Parametrize Ip version
+
+    Args:
+        request: pytest request object
+
+    Returns:
+        Ip version needed for test case
+    """
+    return request.param
+
+
+@pytest.fixture(scope="module", params=["add", "del"])
+def op(request):
+    """
+    Parametrize operate tpye
+
+    Args:
+        request: pytest request object
+
+    Returns:
+        Operate tpye
+    """
+    return request.param
+
+
+class TestConfigDhcpRelay(object):
     def test_plugin_registration(self):
         cli = mock.MagicMock()
         dhcp_relay.register(cli)
-        cli.commands['vlan'].add_command.assert_called_once_with(dhcp_relay.vlan_dhcp_relay)
 
-    def test_config_vlan_add_dhcp_relay_with_nonexist_vlanid(self):
+    def test_test_config_dhcp_relay_add_del_with_nonexist_vlanid(self, ip_version, op):
```

Comment:
Typo? test_test_config_dhcp_relay_add_del_with_nonexist_vlanid

Reply:
Oh, it's a typo, I have fixed it.